### DISCUSSION TITLE
vim E-117 unknown function pumvisible

### DISCUSSION THREAD

**Discussion by serpro69** [JetBrains/ideavim#399] [Q&A] [answered]
I have the following in my vimrc:
```
✓ ~→ grep -e "pumvisible" ~/.vimrc

inoremap <expr> <CR> pumvisible() ? "\<C-y>" : "\<C-g>u\<CR>"
inoremap <expr> <C-n> pumvisible() ? '<C-n>' : '<C-n><C-r>=pumvisible() ? "\<lt>Down>" : ""<CR>'
inoremap <expr> <M-,> pumvisible() ? '<C-n>' : '<C-x><C-o><C-n><C-p><C-r>=pumvisible() ? "\<lt>Down>" : ""<CR>'
```

After upgrading to 1.8.0 I can't use "Enter" key in insert mode anymore, and getting `vim E-117 unknown function pumvisible` error.

Anyone can suggest how I could fix this without modifying .vimrc? Maybe through .ideavimrc?

**Comment by lippfi** [comment, marked as answer]
IdeaVim supports mappings with expressions since 1.7.0, so the line is no longer ignored
You can add `iunmap <CR>` to your `.ideavimrc` (after the line where you source your `.vimrc`) or create an IDE-specific vimrc (it's a new feature and it's kinda cool)  https://github.com/JetBrains/ideavim#ide-specific-options

**Reply by lippfi** [reply]
p.s. now parsing speed is ok, but if you put ideaVim ignore comment in the else block of your vimrc you can make your IDE to startup even faster
```
if has('ide')
  ...
else
  " ideaVim ignore
    ... your config for vim/nvim
  " ideaVim ignore end
endif
```

**Reply by serpro69** [reply]
Thanks a lot for the answer and the extra tip :)
